src/code/su_mtx.c

```diff
@@ -0,0 +1,361 @@
+    //! matrix to not be a rotation matrix (and indeed not even necessarily invertible).
+    // i20 and f20
+    fixedPoint = (tempY - (axisY * sin)) * 0x10000;
+    mtx->intPart[2][0] = ((u32)fixedPoint >> 0x10) & 0xFFFF;
+    mtx->fracPart[2][0] = fixedPoint & 0xFFFF;
+
+    // [i02, i03] and [f02, f03]
+    fixedPoint = (tempY + (axisY * sin)) * 0x10000;
+    mtx->m[0][1] = fixedPoint; // i03 dirty
+    mtx->intPart[0][3] = 0;    // clean i03
+    mtx->m[2][1] = fixedPoint << 0x10;
+
+    // [i22, i23] and [f22, f23]
+    fixedPoint = ((((1.0f - tempZZ) * cos) + tempZZ) * 0x10000);
+    mtx->m[1][1] = fixedPoint; // i23 dirty
+    mtx->intPart[2][3] = 0;    // clean i23
+    mtx->m[3][1] = fixedPoint << 0x10;
+
+    // i10 and f10
+    fixedPoint = (tempZ - axisZ * sin) * 0x10000;
+    mtx->intPart[1][0] = ((u32)fixedPoint >> 0x10) & 0xFFFF;
+    mtx->fracPart[1][0] = fixedPoint & 0xFFFF;
+
+    // i01 and f01
+    fixedPoint = (tempZ + axisZ * sin) * 0x10000;
+    mtx->intPart[0][1] = ((u32)fixedPoint >> 0x10) & 0xFFFF;
+    mtx->fracPart[0][1] = fixedPoint & 0xFFFF;
+}
+
+/**
+ * Creates a general scale, rotation, translation matrix.
+ *
+ * @note The axis vector is expected to be normalised.
+ *
+ * The transformations are applied in the order scale, rotate, translate: \$f SRT \$f
+ *
+ * The result should be
+ * \f[
+ * \begin{bmatrix}
+ *     s_x((1-a_x^2) \cos \theta + a_x^2) & s_x(a_x a_y (1 - \cos \theta) + a_z \sin \theta) & s_x(a_z a_x (1 - \cos
+ *        \theta) - a_y \sin \theta & 0) \\
+ *     s_y(a_x a_y (1 - \cos \theta) - a_z \sin \theta) & s_y((1-a_y^2) \cos \theta + a_y^2) & s_z(a_y a_z (1 - \cos
+ *        \theta) + a_x \sin \theta) & 0 \\
+ *     s_z(a_z a_x (1 - \cos \theta) + a_y \sin \theta) & s_z(a_y a_z (1 - \cos \theta) - a_x \sin \theta) &
+ *        s_z((1-a_z^2) \cos \theta + a_z^2) & 0 \\
+ *     0 & 0 & 0 & 1
+ * \end{bmatrix} .
+ * \f]
+ *
+ * @warning There is a significant bug in this function, which means it does not actually produce a proper rotation
+ * matrix; see inline for details.
+ *
+ * @param[out] mtx Fixed-point matrix pointer to output to
+ * @param[in] scaleX Scale in the X direction
+ * @param[in] scaleY Scale in the Y direction
+ * @param[in] scaleZ Scale in the Z direction
+ * @param[in] angle angle to rotate about axis
+ * @param[in] axisX X component of axis to rotate about
+ * @param[in] axisY Y component of axis to rotate about
+ * @param[in] axisZ Z component of axis to rotate about
+ * @param[in] translateX X component of translation
+ * @param[in] translateY Y component of translation
+ * @param[in] translateZ Z component of translation
+ *
+ * @remark Original name: probably something like "suMtxMakeSRT", although Animal Forest function is a Tait-Bryan
+ * rotation rather than axis-angle.
+ */
+void Mtx_SetTranslationRotationScaleMtx(Mtx* mtx, f32 scaleX, f32 scaleY, f32 scaleZ, s32 angle, f32 axisX, f32 axisY,
```

Comment:
I don't think it's possible to do that while maintaining consistency with the others (as, indeed, you note in the other comment). I'm also not concerned about it since this function is used once and so badly written that we actively want to discourage people from using it.